### PR TITLE
Add support for --device flags on docker run

### PR DESCRIPTION
**Pull request by omeid**
This pull requests adds support for `--device` flag for docker run.

### CODE REVIEW

File: hooks/command [hunk 1]
```diff
@@ -135,6 +135,19 @@ if plugin_read_list_into_result BUILDKITE_PLUGIN_DOCKER_VOLUMES BUILDKITE_PLUGIN
   done
 fi
 
+# Ensure devices option is an array
```
Comment (toolmantim) [inline hooks/command]: I don't think we need these array checks anymore, because `plugin_read_list_into_result` already checks it and outputs an error.

Comment (omeid) [reply hooks/command]: Thanks for the swift feedback. I am happy to remove this instance of it but perhaps a more through cleanup is a better approach?

Comment (toolmantim) [reply hooks/command]: Sure thing! Let's just go with it.

Comment (omeid) [reply hooks/command]: Cool, can you please confirm that accepting `boolean` for `volumes` is a mistake? so that I can safely remove that check too.

**Comment by omeid** [comment]
@toolmantim Thanks for merging this in, I have created #136 as a follow up on the cleanup.

Cheers

**Pull request merged**